### PR TITLE
Issue #9984: restore verify-no-exception-configs

### PR DESCRIPTION
**Pull request by rnveach**
Issue #9984

### CODE REVIEW

**Comment by rnveach** [comment]
https://app.travis-ci.com/github/checkstyle/checkstyle/jobs/585602189
> ./.ci/validation.sh: line 337: xmlstarlet: command not found

Why does this not fail the build?

**Comment by romani** [comment]
It can continue to execute only if `set -e` is not set.

**Comment by rnveach** [comment]
This is just because of `xmlstarlet fo -D ` ?

>   fo    (or format)    - Format XML document(s)

Not seeing documentation on `-D` yet.

There is `  -q or --quiet        - no error output` if we wish to just ignore the error completely.

**Comment by rnveach** [comment]
Job is working now.

https://app.travis-ci.com/github/checkstyle/checkstyle/jobs/585750256#L157
> CMD=export PULL_REQUEST=12300 && ./.ci/validation.sh verify-no-exception-configs
No Diff detected.
eval of CMD is completed

File: .ci/validation.sh [hunk 1]
```diff
@@ -330,15 +330,24 @@ verify-no-exception-configs)
...
-    | sort | uniq | sed "s/Check$//" > $working_dir/web.txt
-  xmlstarlet sel --net --template -m .//module -v "@name" -n config/checkstyle_checks.xml \
+  xmlstarlet fo -D \
+    -n $working_dir/checks-nonjavadoc-error.xml \
+    | xmlstarlet sel --net --template -m .//module -n -v "@name" \
+    | grep -vE $MODULES_WITH_EXTERNAL_FILES | grep -v "^$" > $working_dir/temp.txt
+  xmlstarlet fo -D \
+     -n $working_dir/checks-only-javadoc-error.xml \
+    | xmlstarlet sel --net --template -m .//module -n -v "@name" \
+    | grep -vE $MODULES_WITH_EXTERNAL_FILES | grep -v "^$" >> $working_dir/temp.txt
```
Comment (rnveach) [inline .ci/validation.sh:340]: Changed this to a temp file to combine the 2. The next command wasn't working for me doing it all on 1 file.

File: .ci/validation.sh [hunk 2]
```diff
@@ -330,15 +330,24 @@ verify-no-exception-configs)
     --no-clobber \
     https://raw.githubusercontent.com/checkstyle/contribution/master/checkstyle-tester/checks-only-javadoc-error.xml
   MODULES_WITH_EXTERNAL_FILES="Filter|ImportControl"
-  xmlstarlet sel --net --template -m .//module -v "@name" \
-    -n $working_dir/checks-nonjavadoc-error.xml -n $working_dir/checks-only-javadoc-error.xml \
-    | grep -vE $MODULES_WITH_EXTERNAL_FILES | grep -v "^$" \
-    | sort | uniq | sed "s/Check$//" > $working_dir/web.txt
-  xmlstarlet sel --net --template -m .//module -v "@name" -n config/checkstyle_checks.xml \
+  xmlstarlet fo -D \
```
Comment (rnveach) [inline .ci/validation.sh:333]: `fo -D` is what is needed to resolve the issue.

**Review by romani — APPROVED** [review]
ok to merge.
We will see how it works in real PR with new Check

**Pull request merged**